### ISSUE TITLE
Design and implement new `StatusPort`

### ISSUE THREAD

**Issue by mossmaurice** [eclipse-iceoryx/iceoryx#982]
## Brief feature description

After [this discussion](https://github.com/eclipse-iceoryx/iceoryx/discussions/818) the conclusion was a `StatusPort` or `BroadcastPorts` is beneficial for #415 and for the introspection topics, in order to prevent any user to `take()` the data and not `release()` which could lead to a out-of-memory error.

## Detailed information

* [ ] Design document for `StatusPort`
* [ ] Implement `StatusPorts`
* [ ] Replace `InterfacePorts` with `StatusPorts`

**Comment by elfenpiff** [comment]
@mossmaurice maybe I missed something but the purpose of the interface port is to forward capro messages for instance to gateway ports.
And the purpose of the StatePort is to deliver some kind of state. How should we be able to replace the InterfacePort with a StatusPort?

And shouldn't it be called StatePort instead of StatusPort?

**Comment by mossmaurice** [comment]
@elfenpiff 
> maybe I missed something but the purpose of the interface port is to forward capro messages for instance to gateway ports.

That's correct. More broadly speaking `InterfacePort`s are used to relay discovery information to gateways. After the `StatusPort`s are available we should consolidate them with both the `InterfacePort`s and the introspection 2.0 by sending the `ServiceRegistry` via a `StatusPort`. Fore more information also see the [design document](https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/design/service-discovery.md#decision). Very likely it will be fleshed out from this issue.

> And the purpose of the StatePort is to deliver some kind of state. How should we be able to replace the InterfacePort with a StatusPort?
> And shouldn't it be called StatePort instead of StatusPort?

I think that's a misunderstanding. I'm talking about [these](https://github.com/eclipse-iceoryx/iceoryx/discussions/818) `StatusPort` or `BroadcastPort`s.

**Comment by mossmaurice** [comment]
Not needed anymore. Closing.
